api/utils/pingconn/pingconn_test.go

```diff
@@ -30,252 +30,283 @@ import (
 	"github.com/gravitational/teleport/api/fixtures"
 )
 
+type pingConn interface {
+	net.Conn
+	WritePing() error
+}
+
 func TestPingConnection(t *testing.T) {
-	t.Run("BufferSize", func(t *testing.T) {
-		nWrites := 10
-		dataWritten := []byte("message")
-
-		for _, tt := range []struct {
-			desc    string
-			bufSize int
-		}{
-			{desc: "Same", bufSize: len(dataWritten)},
-			{desc: "Large", bufSize: len(dataWritten) * 2},
-			{desc: "Short", bufSize: len(dataWritten) / 2},
-		} {
-			t.Run(tt.desc, func(t *testing.T) {
-				r, w := makePingConn(t)
+	connTypes := []struct {
+		name     string
+		makeFunc func(t *testing.T) (pingConn, pingConn)
+	}{
+		{
+			name:     "PingConn",
+			makeFunc: makePingConn,
+		},
+		{
+			name:     "PingTLSConn",
+			makeFunc: makePingTLSConn,
+		},
+	}
 
-				// Write routine
-				errChan := make(chan error, 2)
-				go func() {
-					defer w.Close()
+	for _, connType := range connTypes {
+		t.Run(connType.name, func(t *testing.T) {
```

Comment:
no change to the rest of the tests. the diff would be a lot easier hiding whitespace changes.